src/lib/clk.c

```diff
@@ -110,3 +111,41 @@ void platform_timer_set_delta(struct timer *timer, uint64_t ns)
 	platform_shared_commit(clk_info, sizeof(*clk_info));
 	platform_shared_commit(timer, sizeof(*timer));
 }
+
+#if CONFIG_CAVS_PM_RO
```

Comment:
It's cAVS specific, so it should be in platform/intel/cavs/lib/clk.c.
Nevertheless, we already have universal `select_cpu_clock` method. It's better to utilize it somehow.

Reply:
`set_lpro/hpro` should use what we have in `platform/intel/cavs/lib/clk.c` (ofc you can change them if they are missing something, I used them only for hpro/lpro pre-waiti switching.